_blogposts/compiler/string-literal-types-in-reason.mdx

```diff
@@ -0,0 +1,215 @@
+---
+author: hongbo
+date: "2020-07-28 18:00"
+previewImg:
+category: compiler
+title: Introducing string literal types in BuckleScript version 8.2
+description: |
+  Highlights of our newest changes to the internal representation 
+  and how they will benefit our users.
+---
+
+---
+
+## String literal types in BuckleScript
+
+[String literal types](https://www.typescriptlang.org/docs/handbook/advanced-types.html#string-literal-types) was introduced by TypeScript to model
+JavaScript behavior, it's a relatively new concept since most type systems are runtime encoding agnostic. However, to smooth the user experience when writing
+bindings to existing JS API, we are introducing string literal types which is unique in several behaviors compared with TypeScript: it supports type inference,
+pattern match and can be attached to data.
+
+## Vanilla string literal types
```

Comment:
I'd start with a brief refresher how polymorphic variants work. And after that explain how they compiled to javascript before, and then how they compile to javascript in the new situation, with some examples how interop is now better.

Reply:
It's intentionally written this way to have mostly used features easier to understand. There's further reading section for people who want to know more

Reply:
I would contend that this is easier for beginners to introduce a new term that is made up in this blog post and that isn't used anywhere else (not very googleable).

Reply:
Variants and Polymorphic Variants are the main building block of the language. So they should be called out and explained. This is just **very confusing**, both for newcomers and for people familiar with the language because now things are suddenly called something different.